WordPress ESLint runner passes Markdown files from changed-glob scope to ESLint

## Problem

The WordPress extension lint runner sends non-JS files matched by `HOMEBOY_LINT_GLOB` to ESLint.

This breaks docs-only Data Machine PRs when `homeboy lint --changed-since origin/main` resolves the changed set to a Markdown file.

Observed on Extra-Chill/data-machine#1572:

```text
Linting 1 files matching: /Users/chubes/Developer/data-machine@agents-api-extraction-map/docs/development/agents-api-extraction-map.md
...
ESLINT SUMMARY: 1 errors, 0 warnings

/Users/chubes/Developer/data-machine@agents-api-extraction-map/docs/development/agents-api-extraction-map.md
  1:0  error  Parsing error: Unexpected token (1:0)
```

PHPCS passed and PHPStan skipped correctly. Only ESLint failed, because it attempted to parse Markdown as JavaScript.

## Root cause

`wordpress/scripts/lint/eslint-runner.sh` handles non-JS correctly in single-file mode:

```bash
if [ -n "${HOMEBOY_LINT_FILE:-}" ]; then
    ...
    case "${HOMEBOY_LINT_FILE}" in
        *.js|*.jsx|*.ts|*.tsx)
            echo "Linting single file: ${HOMEBOY_LINT_FILE}"
            ;;
        *)
            exit 0
            ;;
    esac
```

But the glob path does not filter by extension:

```bash
elif [ -n "${HOMEBOY_LINT_GLOB:-}" ]; then
    ...
    eval 'for f in '"${HOMEBOY_LINT_GLOB}"'; do [ -e "$f" ] && MATCHED_FILES+=("$f"); done'
    ...
    LINT_FILES=("${MATCHED_FILES[@]}")
```

So a changed-set glob containing only `docs/development/*.md` still becomes an ESLint target.

## Expected behavior

When `HOMEBOY_LINT_GLOB` is set, ESLint should only receive files ending in:

- `.js`
- `.jsx`
- `.ts`
- `.tsx`

If the glob matches files but none are JS/TS files, ESLint should exit 0 with a clear skip message, matching the single-file behavior.

## Suggested fix

Filter `MATCHED_FILES` into a JS-only array before assigning `LINT_FILES`.

Pseudo-shape:

```bash
JS_FILES=()
for f in "${MATCHED_FILES[@]}"; do
    case "$f" in
        *.js|*.jsx|*.ts|*.tsx) JS_FILES+=("$f") ;;
    esac
done

if [ ${#JS_FILES[@]} -eq 0 ]; then
    echo "No JS files match pattern: ${HOMEBOY_LINT_GLOB}"
    exit 0
fi

LINT_FILES=("${JS_FILES[@]}")
```

## Verification target

From a WordPress component with a changed Markdown doc only:

```bash
homeboy lint data-machine --path /Users/chubes/Developer/data-machine@agents-api-extraction-map --changed-since origin/main
```

Expected result: PHPCS passes or no-ops, ESLint skips, PHPStan skips, aggregate lint passes.

## AI assistance

- **AI assistance:** Yes
- **Tool(s):** OpenCode (openai/gpt-5.5)
- **Used for:** Diagnosed the lint-routing failure and drafted this bug report. Chris remains responsible for review and merge decisions.

Additional repro on PHP files from Data Machine PR #1570: https://github.com/Extra-Chill/data-machine/pull/1570

CI job: https://github.com/Extra-Chill/data-machine/actions/runs/25086011241/job/73501621856

Command in CI:

```bash
homeboy --output /tmp/tmp.pmzYDhteCT/lint.json lint data-machine \
  --path /home/runner/work/data-machine/data-machine \
  --changed-since 67bb09cfd426731edf85c5841cf903bd364d02f8
```

The WordPress lint runner correctly scopes the changed set to 6 PHP files and PHPCS passes:

```text
Linting 6 files matching: {...AdjacentHandlerToolSource.php,...ToolSourceRegistry.php,...tests/*-smoke.php}
PHP compatibility target: 8.2-
Validating with PHPCS...
PHPCS linting passed
```

Then the ESLint runner receives the same PHP-file set and fails parsing `<?php`:

```text
ESLINT SUMMARY: 2 errors, 4 warnings

/home/runner/work/data-machine/data-machine/inc/Engine/AI/Tools/Sources/AdjacentHandlerToolSource.php
  1:1  error  Parsing error: Unexpected token (1:1)

/home/runner/work/data-machine/data-machine/inc/Engine/AI/Tools/ToolSourceRegistry.php
  1:1  error  Parsing error: Unexpected token (1:1)

/home/runner/work/data-machine/data-machine/tests/adjacent-handler-tool-policy-smoke.php
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override
```

Local repro from the same branch produces the same output:

```bash
homeboy lint data-machine \
  --path /Users/chubes/Developer/data-machine@untangle-adjacent-handler-tool-source \
  --changed-since origin/main
```

So the extension should filter the changed-glob match list to JS/TS before invoking ESLint, not just Markdown. This is the same root cause as the issue body, with PHP files instead of docs.

--- a/tests/wordpress-eslint-scope-smoke.sh
+++ b/tests/wordpress-eslint-scope-smoke.sh
@@ -33,7 +33,7 @@ COMPONENT_DIR="$TMP_DIR/example-plugin"
 FAKE_EXTENSION="$TMP_DIR/fake-wordpress-extension"
 ESLINT_ARGS_FILE="$TMP_DIR/eslint-args.txt"
 
-mkdir -p "$COMPONENT_DIR/inc" "$COMPONENT_DIR/assets" "$FAKE_EXTENSION/node_modules/.bin"
+mkdir -p "$COMPONENT_DIR/inc" "$COMPONENT_DIR/assets" "$COMPONENT_DIR/docs" "$FAKE_EXTENSION/node_modules/.bin"
 
 cat > "$COMPONENT_DIR/example-plugin.php" <<'PHP'
 <?php
@@ -58,6 +58,10 @@ cat > "$COMPONENT_DIR/assets/view.ts" <<'TS'
 const examplePluginView: boolean = true;
 TS
 
+cat > "$COMPONENT_DIR/docs/example.md" <<'MD'
+# Example docs
+MD
+
 cat > "$FAKE_EXTENSION/.eslintrc.json" <<'JSON'
 {}
 JSON
@@ -65,8 +69,29 @@ JSON
 cat > "$FAKE_EXTENSION/node_modules/.bin/eslint" <<'SH'
 #!/usr/bin/env bash
 printf '%s\n' "$@" > "$ESLINT_ARGS_FILE"
+for arg in "$@"; do
+    case "$arg" in
+        */assets/bad.js|assets/bad.js)
+            echo "simulated eslint failure" >&2
+            exit 1
+            ;;
+    esac
+done
 SH
 chmod +x "$FAKE_EXTENSION/node_modules/.bin/eslint"
+
+HOMEBOY_EXTENSION_PATH="$FAKE_EXTENSION" \
+HOMEBOY_COMPONENT_PATH="$COMPONENT_DIR" \
+HOMEBOY_COMPONENT_ID="example-plugin" \
+ESLINT_ARGS_FILE="$ESLINT_ARGS_FILE" \
+HOMEBOY_LINT_FILE='docs/example.md' \
+    bash "$RUNNER" > "$TMP_DIR/single-md.out" 2>&1
+
+if [ -f "$ESLINT_ARGS_FILE" ]; then
+    echo "ESLint should not run for single-file Markdown scope" >&2
+    cat "$ESLINT_ARGS_FILE" >&2
+    exit 1
+fi
 
 HOMEBOY_EXTENSION_PATH="$FAKE_EXTENSION" \
 HOMEBOY_COMPONENT_PATH="$COMPONENT_DIR" \
@@ -94,5 +119,42 @@ assert_contains "$ESLINT_ARGS_FILE" "assets/admin.js"
 assert_contains "$ESLINT_ARGS_FILE" "assets/view.ts"
 assert_not_contains "$ESLINT_ARGS_FILE" "example-plugin.php"
 assert_not_contains "$ESLINT_ARGS_FILE" "inc/runtime.php"
+
+rm -f "$ESLINT_ARGS_FILE"
+
+HOMEBOY_EXTENSION_PATH="$FAKE_EXTENSION" \
+HOMEBOY_COMPONENT_PATH="$COMPONENT_DIR" \
+HOMEBOY_COMPONENT_ID="example-plugin" \
+ESLINT_ARGS_FILE="$ESLINT_ARGS_FILE" \
+HOMEBOY_LINT_FILE='assets/admin.js' \
+    bash "$RUNNER" > "$TMP_DIR/js-success.out" 2>&1
+
+assert_contains "$TMP_DIR/js-success.out" "Linting single file: assets/admin.js"
+assert_contains "$TMP_DIR/js-success.out" "ESLint linting passed"
+assert_contains "$ESLINT_ARGS_FILE" "assets/admin.js"
+
+cat > "$COMPONENT_DIR/assets/bad.js" <<'JS'
+const bad = true;
+JS
+
+set +e
+HOMEBOY_EXTENSION_PATH="$FAKE_EXTENSION" \
+HOMEBOY_COMPONENT_PATH="$COMPONENT_DIR" \
+HOMEBOY_COMPONENT_ID="example-plugin" \
+ESLINT_ARGS_FILE="$ESLINT_ARGS_FILE" \
+HOMEBOY_LINT_FILE='assets/bad.js' \
+    bash "$RUNNER" > "$TMP_DIR/js-failure.out" 2>&1
+failure_status=$?
+set -e
+
+if [ "$failure_status" -eq 0 ]; then
+    echo "Expected ESLint failure to propagate for JS file scope" >&2
+    cat "$TMP_DIR/js-failure.out" >&2
+    exit 1
+fi
+
+assert_contains "$TMP_DIR/js-failure.out" "Linting single file: assets/bad.js"
+assert_contains "$TMP_DIR/js-failure.out" "ESLint linting failed"
+assert_contains "$ESLINT_ARGS_FILE" "assets/bad.js"
 
 echo "wordpress eslint scope smoke passed"

--- a/wordpress/scripts/lint/eslint-glob-filter-smoke.sh
+++ b/wordpress/scripts/lint/eslint-glob-filter-smoke.sh
@@ -63,7 +63,7 @@ HOMEBOY_LINT_GLOB="{${component_dir}/inc/Thing.php,${component_dir}/assets/app.j
 ESLINT_LOG="$eslint_log" \
     bash "${EXTENSION_PATH}/scripts/lint/eslint-runner.sh" > "${TMPDIR}/mixed.out"
 
-assert_contains "${TMPDIR}/mixed.out" "Linting 1 JS files matching"
+assert_contains "${TMPDIR}/mixed.out" "Linting 1 JS/TS files matching"
 assert_contains "$eslint_log" "${component_dir}/assets/app.js"
 assert_not_contains "$eslint_log" "${component_dir}/inc/Thing.php"
 

